studio/components/to-be-cleaned/Storage/StorageExplorer/PreviewPane.tsx

```diff
@@ -104,104 +113,200 @@ const PreviewPane = () => {
-                  type="default"
-                  icon={<IconDownload size={16} strokeWidth={2} />}
+                  type="outline"
+                  icon={<IconClipboard size={16} strokeWidth={2} />}
+                  onClick={async () => await copyFileURLToClipboard(file)}
                   disabled={file.isCorrupted}
                 >
-                  Download
+                  Copy URL
                 </Button>
-              </a>
-            </Link>
+              ) : (
+                <Dropdown
+                  side="bottom"
+                  align="center"
+                  overlay={[
+                    <Dropdown.Item
+                      onClick={async () => await copyFileURLToClipboard(file, 60 * 60 * 24 * 7)}
+                    >
+                      Expire in 1 week
+                    </Dropdown.Item>,
+                    <Dropdown.Item
+                      onClick={async () => await copyFileURLToClipboard(file, 60 * 60 * 24 * 30)}
+                    >
+                      Expire in 1 month
+                    </Dropdown.Item>,
+                    <Dropdown.Item
+                      onClick={async () => await copyFileURLToClipboard(file, 60 * 60 * 24 * 365)}
+                    >
+                      Expire in 1 year
+                    </Dropdown.Item>,
+                    <Dropdown.Item onClick={() => setShowExpiryInput(true)}>
+                      Custom expiry
+                    </Dropdown.Item>,
+                  ]}
+                >
+                  <Button
+                    type="outline"
+                    icon={<IconClipboard size={16} strokeWidth={2} />}
+                    iconRight={<IconChevronDown />}
+                    disabled={file.isCorrupted}
+                  >
+                    Get Signed URL
+                  </Button>
+                </Dropdown>
+              )}
+            </div>
             <Button
               type="outline"
-              icon={<IconClipboard size={16} strokeWidth={2} />}
-              onClick={async () => await copyFileURLToClipboard(file)}
-              disabled={file.isCorrupted}
+              shadow={false}
+              size="tiny"
+              icon={<IconTrash2 size={16} strokeWidth={2} />}
+              onClick={() => setSelectedItemsToDelete([file])}
             >
-              Copy URL
+              Delete file
             </Button>
           </div>
-          <Button
-            type="outline"
-            shadow={false}
-            size="tiny"
-            icon={<IconTrash2 size={16} strokeWidth={2} />}
-            onClick={() => setSelectedItemsToDelete([file])}
-          >
-            Delete file
-          </Button>
         </div>
-      </div>
-    </Transition>
+      </Transition>
+      <Modal
+        hideFooter
+        size="small"
+        header="Custom expiry for signed URL"
+        visible={showExpiryInput}
+        alignFooter="right"
+        confirmText="Get signed URL"
```

Comment:
I just realized what a "signed url" means!  How have I gone this long! 

I'm wondering in that case, if this should use some other language to convey it.  Something like "Get a signed (expiring) url".  Or a little note somewhere there — a tooltip maybe?

Reply:
actually tbh now that i think about it the UI probably doesnt need to discern between a URL and a signed URL perhpas
just that URLs from private buckets have expiry but URLs from public buckets do not

the dropdown which follows the language "expires in xxx" probably conveys the expiration clear enough